tests/server/test_environement_metrics.py

```diff
@@ -339,3 +342,262 @@ async def get_metric_value(
+            "send_event": False,
+            "purged": False,
+            "requires": [],
+        },
+        {
+            "key": "key6",
+            "value": "value6",
+            "id": "test::Resource[agent1,key=key6],v=" + version_env2,
+            "send_event": False,
+            "requires": [],
+            "purged": False,
+        },
+    ]
+    result = await client.put_version(
+        tid=env_uuid2,
+        version=version_env2,
+        resources=resources_env2,
+        unknowns=[],
+        version_info={},
+        compiler_version=get_compiler_version(),
+    )
+    assert result.code == 200
+    assert len(await data.Resource.get_list()) == 6
+
+    # adds the ResourceCountMetricsCollector
+    rcmc = ResourceCountMetricsCollector()
+    metrics_service.register_metric_collector(metrics_collector=rcmc)
+
+    # flush the metrics for the first time: 2 record (3 resources in available state for the first
+    # environment and 2 for the second)
+    await metrics_service.flush_metrics()
+    result_gauge = await data.EnvironmentMetricsGauge.get_list()
+    assert len(result_gauge) == 2
+    assert any(
+        x.count == 3 and x.metric_name == "resource_count.available" and x.environment == env_uuid1 for x in result_gauge
+    )
+    assert any(
+        x.count == 2 and x.metric_name == "resource_count.available" and x.environment == env_uuid2 for x in result_gauge
+    )
+
+    # change the state of one of the resources
+    now = datetime.now()
+    action_id = uuid.uuid4()
+    aclient = agent._client
+    result = await aclient.resource_action_update(
+        env_uuid1,
+        ["test::Resource[agent1,key=key2],v=" + version_env1],
+        action_id,
+        "deploy",
+        now,
+        now,
+        "deployed",
+        [],
+        {},
+    )
+
+    assert result.code == 200
+
+    # flush the metrics for the second time: 2 old record +
+    # + 3 new records (1 for available state and one for the deployed state for the first environment
+    # and one for the available state for the second environment)
+    await metrics_service.flush_metrics()
+    result_gauge = await data.EnvironmentMetricsGauge.get_list()
+    assert len(result_gauge) == 5
+    assert any(
+        x.count == 3 and x.metric_name == "resource_count.available" and x.environment == env_uuid1 for x in result_gauge
+    )
+    assert any(
+        x.count == 2 and x.metric_name == "resource_count.available" and x.environment == env_uuid1 for x in result_gauge
+    )
+    assert any(x.count == 1 and x.metric_name == "resource_count.deployed" and x.environment == env_uuid1 for x in result_gauge)
+
+    env_uuid2_records = [
+        r for r in result_gauge if r.environment == env_uuid2 and r.metric_name == "resource_count.available" and r.count == 2
+    ]
+
+    assert len(env_uuid2_records) == 2
+
+
+async def test_resource_count_metric_released(clienthelper, client, agent):
```

Comment:
This is implied by using the client fixture, but I like it when this is made explicit.

```suggestion
async def test_resource_count_metric_released(clienthelper, client, server, agent):
```